modules/api_gateway/main.tf

```diff
@@ -43,7 +43,8 @@ locals {
   level_5_routes = { for key, value in local.routes : key => value if length(split("/", key)) == 5 }
   redeployment_hash = sha1(jsonencode(concat(
     [
-      var.name
+      var.name,
+      var.redeployment_hash
```

Comment:
Should this have a different name than the same key being passed to the api gateway module? Is there any reason to have the lists below and not force just using the redeployment hash?

Reply:
One being a `var` and the other being a `local` is enough of a distinction for me but I don't hold that opinion strongly. I'm open to suggestions on a different variable name.

Reply:
The scenario that led to these changes was nothing changed and a deploy happened to something unspecified. I would rather the redeployment_hash just be passed straight through to the original terraform module.

There are two mechanisms that might trigger a redeployment here and they seem to both change at the same time. Most recommendations were to pass in a hash. If you don't want the deployment to happen, you can't use this module because changes could cause a redeployment and only guaranteed if using the redeployment hash.

Reply:
Point taken.